src/packet.ml

```diff
@@ -80,47 +79,39 @@ type header = {
 }
 
 let pp_header ppf hdr =
-  Fmt.pf ppf "local %Lu replay_id %ld timestamp %ld hmac %a ack %a remote %a"
-    hdr.local_session hdr.replay_id hdr.timestamp Cstruct.hexdump_pp hdr.hmac
+  Fmt.pf ppf "local %Lu replay_id %ld timestamp %ld ack %a remote %a"
+    hdr.local_session hdr.replay_id hdr.timestamp
     Fmt.(list ~sep:(any ", ") uint32)
     hdr.ack_sequence_numbers
     Fmt.(option ~none:(any " ") uint64)
     hdr.remote_session
 
 let header = function `Ack hdr | `Control (_, (hdr, _, _)) -> hdr
 
-let decode_header ~hmac_len buf =
+let decode_header buf =
   let open Result.Syntax in
-  let* () = guard (Cstruct.length buf >= hdr_len hmac_len) `Partial in
-  let local_session = Cstruct.BE.get_uint64 buf 0
-  and hmac = Cstruct.sub buf 8 hmac_len
-  and replay_id = Cstruct.BE.get_uint32 buf (hmac_len + 8)
-  and timestamp = Cstruct.BE.get_uint32 buf (hmac_len + 12)
-  and arr_len = Cstruct.get_uint8 buf (hmac_len + 16) in
+  (* our input is a buffer where the hmac has already been swapped away *)
```

Comment:
maybe this needs to be more prominent?

Reply:
maybe we can add a `type hmac_swapped = private Cstruct.t` and have `split_hmac` return `Cstruct.t * hmac_swapped`

Reply:
hmm, I'm not too convinced of adding more types... esp. with the bytes/string in sight, I wouldn't mess around with it.. there's now the asymmetry between encode_header and decode_header - which is fine, in respect to the comment... alternatively we can rename decode_header to decode_hmac_split_header or so..